tests/gradients/test_parameter_shift_cv.py

```diff
@@ -0,0 +1,1105 @@
+        finite difference and analytic methods."""
+        tol = 1e-2
+
+        args = np.linspace(0.2, 0.5, op.num_params)
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Displacement(0.5, 0, wires=0)
+            op(*args, wires=range(op.num_wires))
+            qml.Beamsplitter(1.3, -2.3, wires=[0, 1])
+            qml.Displacement(-0.5, 0.1, wires=0)
+            qml.Squeezing(0.5, -1.5, wires=0)
+            qml.Rotation(-1.1, wires=0)
+            qml.var(obs(wires=0))
+
+        dev = qml.device("default.gaussian", wires=2)
+        res = tape.execute(dev)
+
+        tape.trainable_params = set(range(2, 2 + op.num_params))
+
+        # jacobians must match
+        tapes, fn = qml.gradients.finite_diff(tape)
+        grad_F = fn(dev.batch_execute(tapes))
+
+        tapes, fn = param_shift_cv(tape, dev)
+        grad_A = fn(dev.batch_execute(tapes))
+
+        tapes, fn = param_shift_cv(tape, dev, force_order2=True)
+        grad_A2 = fn(dev.batch_execute(tapes))
+
+        assert np.allclose(grad_A2, grad_F, atol=tol, rtol=0)
+        assert np.allclose(grad_A, grad_F, atol=tol, rtol=0)
+
+        # check that every parameter is analytic
+        if obs != qml.NumberOperator:
+            for i in range(op.num_params):
+                assert tape._par_info[2 + i]["grad_method"][0] == "A"
+
+    def test_squeezed_mean_photon_variance(self, tol):
+        """Test gradient of the photon variance of a displaced thermal state"""
+        dev = qml.device("default.gaussian", wires=1)
+
+        r = 0.12
+        phi = 0.105
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Squeezing(r, 0, wires=0)
+            qml.Rotation(phi, wires=0)
+            qml.var(qml.X(wires=[0]))
+
+        tape.trainable_params = {0, 2}
+        tapes, fn = param_shift_cv(tape, dev)
+        grad = fn(dev.batch_execute(tapes))
+        expected = np.array(
+            [
+                2 * np.exp(2 * r) * np.sin(phi) ** 2 - 2 * np.exp(-2 * r) * np.cos(phi) ** 2,
+                2 * np.sinh(2 * r) * np.sin(2 * phi),
+            ]
+        )
+        assert np.allclose(grad, expected, atol=tol, rtol=0)
+
+    def test_displaced_thermal_mean_photon_variance(self, tol):
+        """Test gradient of the photon variance of a displaced thermal state"""
+        dev = qml.device("default.gaussian", wires=1)
+
+        n = 0.12
+        a = 0.105
+
+        with qml.tape.JacobianTape() as tape:
+            qml.ThermalState(n, wires=0)
+            qml.Displacement(a, 0, wires=0)
+            qml.var(qml.TensorN(wires=[0]))
+
+        tape.trainable_params = {0, 1}
+        tapes, fn = param_shift_cv(tape, dev)
+        grad = fn(dev.batch_execute(tapes))
+        expected = np.array([2 * a ** 2 + 2 * n + 1, 2 * a * (2 * n + 1)])
+        assert np.allclose(grad, expected, atol=tol, rtol=0)
+
+
+class TestParamShiftInterfaces:
+    """Test that the transform is differentiable"""

```

Comment:
I know I asked the other day, but could you please remind me the purpose of the differentiation tests and how they work when the PR states explicitly that the transform is not differentiable?

Reply:
Ah! So. All my intuition says that the transform should not be differentiable, because so much of it uses vanilla numpy.

I wanted to write these tests anyway, and just mark them `xfail`, so we had tests ready in case we needed them in the future (and to remind us that this doesn't work).

But, the tests.... worked????

Against all my intuition.

I don't really trust them though, I am sure there are many edge cases where autodiff would fail... I just can't find them haha